--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -109,6 +109,10 @@
             return false;
         }
         const peerRecord = PeerRecord.createFromProtobuf(envelope.payload);
+        if (!peerRecord.peerId.equals(peerId)) {
+            this.log('signing key did not match peer id in the peer record - signer: %p record: %p', peerId, peerRecord.peerId);
+            return false;
+        }
         let peer;
         try {
             peer = await this.get(peerId, options);
--- a/src/index.ts
+++ b/src/index.ts
@@ -183,6 +183,12 @@
     }
 
     const peerRecord = PeerRecord.createFromProtobuf(envelope.payload)
+
+    if (!peerRecord.peerId.equals(peerId)) {
+      this.log('signing key did not match peer id in the peer record - signer: %p record: %p', peerId, peerRecord.peerId)
+      return false
+    }
+
     let peer: Peer | undefined
 
     try {
